### PR TITLE
S1 user buttons

### PR DESCRIPTION
**Pull request by ychoy**

### CODE REVIEW

File: app/controllers/users_controller.rb [hunk 1]
```diff
@@ -37,6 +37,19 @@ def update
     end
   end
 
+  def destroy
+    if current_user
+      @user = User.find(params[:id])
+      @user.tips.destroy
```
Comment (nathanallen) [inline app/controllers/users_controller.rb]: aka "dependent destroy"
http://guides.rubyonrails.org/association_basics.html#options-for-belongs-to

File: app/models/user.rb [hunk 2]
```diff
@@ -1,7 +1,7 @@
 class User < ApplicationRecord
   has_secure_password
 
-  has_many :tips
+  has_many :tips, dependent: :destroy
```
Comment (nathanallen) [inline app/models/user.rb]: missing reciprocal dependent destroy.

File: app/views/cities/show.html.erb [hunk 3]
```diff
@@ -1,6 +1,13 @@
...
 <img class='banner-img' src=<%= @city.photo%>>
-<%= button_to "Add New Tip", new_tip_path, method: "get"%>
-<% @city.tips.each do |tip|%>
-  <%= link_to tip.title, tip_path(tip) %> <br>
-<% end %>
+<br>
+<br>
+<div class="container">
+  <%= button_to "Add New Tip", new_tip_path, method: "get"%>
+  <% @city.tips.order(created_at: :desc).each do |tip|%>
```
Comment (nathanallen) [inline app/views/cities/show.html.erb]: Is this the cities#show page or the tips#index page for a specific city? If you think of it as being a tips#index you may be able to move some of this logic into a different view, a view that only knows about "locations" and not "cities".

File: config/routes.rb [hunk 4]
```diff
@@ -15,6 +15,8 @@
 
   patch "/users/:id", to: "users#update", as: "update_user"
 
+  delete "/users/:id", to: "users#destroy", as: "destroy_user"
```
Comment (nathanallen) [inline config/routes.rb]: No path alias needed. The path here is the same as the path for users#show, so use the user_path instead.

File: app/views/cities/index.html.erb [hunk 5]
```diff
@@ -1,6 +1,10 @@
-<div class="container">
-  <%= image_tag("lvg_yanyinchoy.png", width: "100%", class:"img-fluid", alt:"homepage photo") %>
+<div class="jumbotron">
+
+    <%= image_tag("lvg_yanyinchoy.png", width: "100%", class:"img-fluid", alt:"homepage photo") %>
```
Comment (cnwilliams) [inline app/views/cities/index.html.erb:3]: spaces

**Pull request merged**